src/test/integration/integrationtests.py

```diff
@@ -145,14 +156,44 @@ def test_parking(self):
         # Test that a multi-query-parameter request for parkingZoneGroup
         # only returns parking locations that match one of the specified zones
         parking_zones = set(['A1', 'C', 'B2'])
-        multi_zone_query = query_request(locations_url, access_token, "get",
-                {'parkingZoneGroup': parking_zones, 'campus': 'corvallis', 'page[size]': max_page_size}).json()
+        multi_zone_query = query_request(locations_url, access_token, "get", {
+            'parkingZoneGroup': parking_zones, 'campus': 'corvallis', 'page[size]': max_page_size}).json()
 
-        result_parking_zones = set([parking_location['attributes']['parkingZoneGroup']
-            for parking_location in multi_zone_query['data']])
+        result_parking_zones = set([parking_location['attributes']['parkingZoneGroup'] for parking_location in multi_zone_query['data']])
 
         self.assertEqual(parking_zones, result_parking_zones)
 
+    # Test results returned with isOpen=true are actually open given what their openHours say
+    def test_isopen(self):
+        def test_resource(resource_url):
+            all_open_resources = query_request(
+                resource_url,
+                access_token,
+                'get',
+                {'page[size]': max_page_size, 'isOpen': 'true'}
+            ).json()
+
+            now = datetime.utcnow().replace(microsecond=0).isoformat()
+            weekday = str(datetime.today().weekday() + 1)
+
+            for open_resource in all_open_resources['data']:
+                # Test that only open resources are returned when they should be and each open resource has a related open hours
+                open_hours = open_resource['attributes']['openHours']
+                self.assertIsNotNone(open_hours)
+
+                # If there are not only one open hours, current time should be in one of these open hours intervals
+                if len(open_hours[weekday]) > 1:
+                    is_open = False
+                    for open_hour in open_hours[weekday]:
+                        if open_hour['start'][:-1] < now < open_hour['end'][:-1]:
+                            is_open = True
+                            self.assertTrue(is_open)
```

Comment:
What if none of the time slots contain the current time? This test would never fail, it just wouldn't run, yes?

Reply:
yes. i think this line needs to be dedented by a couple blocks. it seems like what we really want to say is `self.assertTrue( any(open_hour['start'] <= now+'Z' <= open_hour['end'] for open_hour in open_hours[weekday]) )`

Reply:
@jaredkosanovic good catch. Will fix it.